### PR TITLE
Refactor AnimationStateFlag handling

### PR DESCRIPTION
**Pull request by xezon**
A few minor improvements to w3dmodeldraw.

### CODE REVIEW

File: src/game/client/draw/w3dmodeldraw.cpp [hunk 1]
```diff
@@ -144,7 +144,7 @@ void ModelConditionInfo::Validate_Stuff(RenderObjClass *robj, float scale, std::
     Validate_Weapon_Barrel_Info();
 }
 
-bool Test_Animation_Flag(int flags, char flag)
+bool Test_Animation_Flag(int flags, AnimationStateFlag flag)
```
Comment (xezon) [inline src/game/client/draw/w3dmodeldraw.cpp:147]: This is more explicit. No need for it to be a `char` when we pass `AnimationStateFlag` to it.

File: src/game/client/draw/w3dmodeldraw.cpp [hunk 2]
```diff
@@ -1472,14 +1472,21 @@ const ModelConditionInfo *W3DModelDraw::Find_Transition_For_Sig(uint64_t sig) co
...
 bool Maintain_Frame_Across_States(int flags)
 {
-    return (flags & 0x3A0) != 0;
+    return (flags & g_maintain_frame_across_states_flags) != 0;
 }
 
 bool Maintain_Frame_Across_States_2(int flags, int flags2)
 {
-    return (flags2 & 0x3A0 & flags & 0x3A0) != 0;
+    return (flags2 & flags & g_maintain_frame_across_states_flags) != 0;
```
Comment (xezon) [inline src/game/client/draw/w3dmodeldraw.cpp:1489]: No need to mask it twice here. Once will do.

File: src/game/client/draw/w3dmodeldraw.cpp [hunk 3]
```diff
@@ -1502,7 +1509,7 @@ float W3DModelDraw::Get_Current_Anim_Fraction() const
     }
 
     if (frames > frame) {
-        return frame / (frames - 1.0f);
+        return frame / (frames - 1);
```
Comment (xezon) [inline src/game/client/draw/w3dmodeldraw.cpp]: `frames` is integer, so can deduct integer.

**Review by xezon — COMMENTED** [review]
I tested and animations looked as usual.

File: src/game/client/draw/w3dmodeldraw.cpp (hunk 3 above)
Comment (xezon) [reply src/game/client/draw/w3dmodeldraw.cpp]: Reverted.

**Comment by tomsons26** [comment]
change if ((m_curState->m_flags & 1 << ADJUST_HEIGHT_BY_CONSTRUCTION_PERCENT) != 0) { to Test_Animation_Flag as well while you are at it
and check if there's any other direct uses of it and do the same, imo it shouldn't be directly checked like this at all

**Comment by xezon** [comment]
Fixed. I did not see any others.

**Review by tomsons26 — APPROVED** [review]

**Pull request merged**